src/ai/backend/common/events.py

```diff
@@ -221,70 +222,108 @@ def deserialize(cls, value: tuple):
 class ImagePullStartedEvent(AbstractEvent):
     name = "image_pull_started"
 
-    image: str = attrs.field()
+    image: str = attrs.field()  # deprecated, use image_ref
     agent_id: AgentId = attrs.field()
     timestamp: float = attrs.field()
+    image_ref: Optional[ImageRef] = attrs.field(default=None)
 
     def serialize(self) -> tuple:
+        if self.image_ref is None:
+            return (self.image, str(self.agent_id), self.timestamp)
+
         return (
             self.image,
             str(self.agent_id),
             self.timestamp,
+            self.image_ref,
         )
 
     @classmethod
     def deserialize(cls, value: tuple):
+        # Backward compatibility
+        if len(value) <= 3:
+            return cls(
+                image=value[0],
+                agent_id=AgentId(value[1]),
+                timestamp=value[2],
+            )
+
```

Comment:
I really don't like this approach, but I can't think of another way since `value` is tuple...

Reply:
FYI, compatibility with the legacy agent has been confirmed. @HyeockJinKim